### PR TITLE
chore: remove benchmark feature

### PR DESCRIPTION
**Pull request by joske**

### CODE REVIEW

**Review by ljedrz — APPROVED** [review]
LGTM in general, gonna check those delays locally :detective:

**Comment by ljedrz** [comment]
While I can see how removing some benchmark-related code might (positively) impact the performance of some tests, I can't reproduce any test issues (even when running in debug mode) with `test_primary_no_consensus` or `test_primary_with_consensus`.

**Review by niklaslong — APPROVED** [review]

**Comment by niklaslong** [comment]
Looks good to me but I can't reproduce the timeout issue either (on M1).

File: narwhal/node/tests/node_smoke_test.rs [hunk 1]
```diff
@@ -50,6 +53,8 @@ fn test_primary_no_consensus() {
     let workers_file_path = format!("{config_path}/smoke_test_workers.json");
     worker_cache.export(&workers_file_path).unwrap();
 
+    thread::sleep(Duration::from_millis(500)); // no idea why this is now needed :-/
```
Comment (ljedrz) [inline narwhal/node/tests/node_smoke_test.rs:56]:
```suggestion
    // TODO: this spot might require a delay if the test is flaky
```

File: narwhal/node/tests/node_smoke_test.rs [hunk 2]
```diff
@@ -134,6 +139,8 @@ fn test_primary_with_consensus() {
...
```
Comment (ljedrz) [inline narwhal/node/tests/node_smoke_test.rs:142]: ditto

**Pull request merged**